src/core/unit.c

```diff
@@ -4162,7 +4162,7 @@ UnitFileState unit_get_unit_file_state(Unit *u) {
 
         assert(u);
 
-        if (u->unit_file_state >= 0 || !u->fragment_path)
```

Comment:
Hmm, this means we'd unnecessarily do file lookup when the unit has no main file? Why? Moreover for perpetual units without fragment we'd start returning _BAD while previously _INVALID, which is reflected on dbus property... Sounds problematic to me and not beneficial to say the least.